### PR TITLE
[IMP] website: make carousel arrows not editable

### PR DESCRIPTION
**Pull request by younn-o**
Added o_not_editable class to carousel-control-prev and
carousel-control-next container divs so that the user can't add text
when sliding carousel items

task-2446852

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website/static/src/js/tours/tour_utils.js
Comment (sdegueldre) [inline addons/website/static/src/js/tours/tour_utils.js:205]: @qsm-odoo does this work now? Translating template strings?

Comment (qsm-odoo) [reply addons/website/static/src/js/tours/tour_utils.js:205]: No, you should use `_.str.sprintf` and if you add translatable terms in standard, don't forget to update the `pot` file :wink:

Comment (sdegueldre) [reply addons/website/static/src/js/tours/tour_utils.js:205]: Well, the tour_utils file uses this all over so that probably needs to be fixed, as you can see this commit isn't the one that introduced this use of `_t`.

Comment (qsm-odoo) [reply addons/website/static/src/js/tours/tour_utils.js:205]: Indeed, maybe @younn-o can fix all of that in the same commit and updating the .pot file. That way that will give a good excuse for actually changing the sentence in a stable version ;)

File: addons/website/static/src/js/menu/new_content.js
Comment (JKE-be) [inline addons/website/static/src/js/menu/new_content.js:34]: " and not '   following framework js spec if i'm not wrong ;)

File: addons/website/static/src/js/menu/new_content.js
Comment (sdegueldre) [inline addons/website/static/src/js/menu/new_content.js:323]: You can use a forEach loop and save yourself some manual book-keeping:

```js
self.newContentText.installPleaseWait[name].forEach((message, i) => {
  setTimeout(..., duration)
}
```

File: addons/website/views/snippets/s_company_team.xml
Comment (sdegueldre) [inline addons/website/views/snippets/s_company_team.xml:11]: This probably doesn't work after saving: the `contenteditable` attributes will be stripped and won't be added back in the next editing session. You probably need the `o_editable` class for that to work as intended, although you might need some additional work to make sure the editor handles nested `o_editable`/`o_not_editable` correctly.

File: addons/website_sale/views/templates.xml
Comment (sdegueldre) [inline addons/website_sale/views/templates.xml:1879]: Doesn't this make the image editable even when you're not in edit-mode? I remember we talked about this but I'm not sure anymore. Also, any reason `o_not_editable` doesn't work on the wrapper?

File: addons/website/static/src/js/editor/snippets.options.js
Comment (sdegueldre) [inline addons/website/static/src/js/editor/snippets.options.js:2424]: `$.closest()` will avoid collecting all the ancestors while searching and will stop at the first element matching the selector.

File: addons/website_blog/models/website_blog.py
Comment (sdegueldre) [inline addons/website_blog/models/website_blog.py:132]: I don't think this is the fix we want. This also adds all the options of sections (including columns and stuff) on the blog-post content.

QSM seemed to be saying that the issue here is that the anchor option targets all children of oe_structure, which causes an option block to appear in the right bar and an overlay to show up. People Shouldn't be able to link to a random line/text block in their blog post in the first place, and adjusting the selector of the anchor option should also make the overlay and option block disappear completely.

**Review by sdegueldre — CHANGES_REQUESTED** [review]

File: addons/web_editor/static/src/scss/wysiwyg_snippets.scss
Comment (qsm-odoo) [reply addons/web_editor/static/src/scss/wysiwyg_snippets.scss:55]: You can remove that ;)

File: addons/website/views/snippets/s_share.xml
Comment (qsm-odoo) [reply addons/website/views/snippets/s_share.xml:6]: Also, even if it was correct, you could not just rename variables in a template in a stable version :wink:

File: addons/website/views/snippets/s_company_team.xml
Comment (qsm-odoo) [reply addons/website/views/snippets/s_company_team.xml:11]: Not sure what is done here, but be careful: o_editable is not a class you should add by yourself. Only the o_not_editable can be added on elements you don't want to be editable and in that case contenteditable="false" would not be necessary... normally :grimacing:

File: addons/website_blog/models/website_blog.py
Comment (qsm-odoo) [reply addons/website_blog/models/website_blog.py:132]: In fact do not care about this, I don't know why it is in the bag of ui fixes task... this is in fact handled by OME in another task ;)

File: addons/website/views/snippets/s_carousel.xml
Comment (younn-o) [reply addons/website/views/snippets/s_carousel.xml:66]: It was changed to div to make it consistent with the s_quotes_carousel controls but I did not think about css that could target <a> tags, I will revert that

File: addons/website/views/snippets/s_company_team.xml
Comment (sdegueldre) [reply addons/website/views/snippets/s_company_team.xml:11]: My understanding is that o_not_editable will only be processed on editor startup, is that not the case?

Comment (qsm-odoo) [reply addons/website/views/snippets/s_company_team.xml:11]: Maybe, I am never sure. In that case, maybe contenteditable false on top of that is ok indeed

File: addons/website/static/src/js/menu/new_content.js
Comment (sdegueldre) [reply addons/website/static/src/js/menu/new_content.js:41]: Nevermind, website_sale is not installed yet so the include wouldn't work... Maybe we could have that data in the manifest and load it into the ir.module, but that's certainly not something we can do in stable. @qsm-odoo opinion?

Comment (qsm-odoo) [reply addons/website/static/src/js/menu/new_content.js:41]: No need to loose time on this, for installation stuff, we can accept stuff like that. However, the implementation is not ok: a translator will not be able to translate parts of sentences individually and have a correct result. In french, the last part already goes

"Vous allez pouvoir"
"Designer des beaux"
"Pages de produits"

... fail. Get a language which is not structured the same way like Dutch and the sentence won't even be understandable. Maybe the spec has to be reviewed, check with the PO.

File: addons/website_sale/views/templates.xml
Comment (younn-o) [reply addons/website_sale/views/templates.xml:1880]: I had to use a container at the size of the image around it and make it not editable in order to prevent writing text at the level of the image. As the image is editable, if it is displayed in a bigger container, it will be possible to write text besides the image

File: addons/website/static/src/js/menu/new_content.js
Comment (younn-o) [reply addons/website/static/src/js/menu/new_content.js]: Answer from the PO :

> I actually think that a good translator should look at the three bits and translate them so that they work together.
> Typically in French you would have;
> 'Dans 20 secs'
> 'Vous allez pouvoir'
> 'Designer votre magasin en ligne'
> 'et y mettre vos superbes produits'

https://www.odoo.com/web#active_id=2446852&cids=1&id=2446852&model=project.task&menu_id=

**Comment by sdegueldre** [comment]
@d-fence ping

File: addons/website/static/src/js/editor/snippets.options.js
Comment (sdegueldre) [inline addons/website/static/src/js/editor/snippets.options.js:2464]: In this case you might be able to simply avoid instanciating this option by adding the right selector to the data-exclude attribute instead of simply hiding it with `_computeWidgetVisibility`

File: addons/website/static/src/js/menu/new_content.js
Comment (sdegueldre) [inline addons/website/static/src/js/menu/new_content.js:17]: looks like a linting error.

File: addons/website/static/src/js/menu/new_content.js
Comment (sdegueldre) [inline addons/website/static/src/js/menu/new_content.js:196]: missing args documentation

**Review by sdegueldre — CHANGES_REQUESTED** [review]
Almost good to go, last few things for me then QSM can take a look for merging :)

File: addons/website_sale/views/templates.xml
Comment (qsm-odoo) [inline addons/website_sale/views/templates.xml:1881]: Let's not make that kind of changes in a stable version. Changing the XML structure is only to fix important bugs, this is not one ;) Move that commit in the master branch.

File: addons/web_editor/static/src/scss/wysiwyg.scss
Comment (qsm-odoo) [inline addons/web_editor/static/src/scss/wysiwyg.scss:128]: Commit message: "make dropdown menus should scrollable " is not proper english ;)

File: addons/website/views/snippets/s_share.xml
Comment (qsm-odoo) [inline addons/website/views/snippets/s_share.xml:6]: Commit message: mentioning website_event for no reason

File: addons/website/i18n/website.pot
Comment (qsm-odoo) [inline addons/website/i18n/website.pot:2895]: Rebase your branch, @JKE-be made a commit changing this sentence last week (without fixing translation like you do here) so there will be a conflict here :wink: But I think JKE use the correct english "at the bottom" and not "in the bottom"... right?

File: addons/website/static/src/js/tours/tour_utils.js
Comment (qsm-odoo) [inline addons/website/static/src/js/tours/tour_utils.js:111]: Don't use `` ` `` for normal strings, it is confusing

File: addons/website/views/snippets/snippets.xml
Comment (qsm-odoo) [inline addons/website/views/snippets/snippets.xml:988]: I am not sure what this is fixing? Seems in conflict with other task, can you explain/link a screenshot?

File: addons/website/static/src/xml/website.editor.xml
Comment (qsm-odoo) [inline addons/website/static/src/xml/website.editor.xml:150]: - New term -> need pot file update for translation
- Dynamic term for a translation... bad idea to just randomly change a word inside.

The easiest way to do that properly: an empty `<p>` here in which you add a full sentence from the JS, translated via the JS using `_.str.sprintf(_t(...), ...)` like you did before.

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

File: addons/website/views/snippets/snippets.xml
Comment (younn-o) [reply addons/website/views/snippets/snippets.xml:988]: It was linked to that specification :

User is not able to change the witdh of the first content of a blog because it's not part of a blog. It's weird for the user that 'content width' buttno have no effect here. https://www.awesomescreenshot.com/video/2549246?key=685df43eccbb8d981623c6b95253805d (just exclude the content width option in blog post -> better will be done in master)

The commit message is confusing, it is in fact the option "container width" of every block dragged into a blog that will be ineffective, thus it is hidden

Before :
image

After :
image

Comment (qsm-odoo) [reply addons/website/views/snippets/snippets.xml:988]: Mmh, indeed. I was confused by another spec for master. Dragging that kind of block in blogs is already not the intented (because ugly) behavior so it feels kinda pointless to "fix" this problem... anyways we can do it if it is in the specs. However: you are fixing a website_blog behavior inside the website app: you should fix it in website_blog (`o_wblog_post_content` is not an ID of a DOM node the website app is supposed to know about). So an xpath in website_blog to add this `data-exclude`.

File: addons/website_sale/views/templates.xml
Comment (younn-o) [reply addons/website_sale/views/templates.xml:1881]: The commit is now part of https://github.com/odoo/odoo/pull/66132

File: addons/website/views/snippets/snippets.xml
Comment (younn-o) [reply addons/website/views/snippets/snippets.xml:988]: Ok, thank you for the explanations, should I move it to this PR for master https://github.com/odoo/odoo/pull/66132 or keep it on this one ?

Comment (qsm-odoo) [reply addons/website/views/snippets/snippets.xml:988]: This one :+1:

File: addons/website/static/src/js/tours/tour_utils.js
Comment (sdegueldre) [inline addons/website/static/src/js/tours/tour_utils.js:138]: @qsm-odoo FYI, after seeing this change I asked MOC if it was possible to add a check against template strings in translations on the runbot, which he made, we're waiting for this PR to be merged before we can fix the existing cases and add that test on the runbot, but hopefully this kind of issue should no longer slip by unnoticed

Comment (d-fence) [reply addons/website/static/src/js/tours/tour_utils.js:138]: @sdegueldre the [branch](https://github.com/odoo-dev/odoo/tree/master-lint-template-strings-moc) is ready, but I cannot merge it until all the template-strings are fixed. That's why I need your help as asked on discord 🤙🏻

Comment (sdegueldre) [reply addons/website/static/src/js/tours/tour_utils.js:138]: @d-fence Yes I understood that, I thought this PR would be be merged quickly and I could fix the other existing cases after that but maybe I can already make a PR with all changes not covered by this one. I'll look at doing that tomorrow then, thanks again for the test, I think it confuses a lot of people :)

File: addons/website_sale/static/src/scss/website_sale.scss
Comment (qsm-odoo) [inline addons/website_sale/static/src/scss/website_sale.scss:77]: What does this do?

**Review by qsm-odoo — CHANGES_REQUESTED** [review]
@younn-o Some last minutes comments. Can you handle them so we can merge afterwards? :+1:

File: addons/website_sale/static/src/scss/website_sale.scss
Comment (younn-o) [reply addons/website_sale/static/src/scss/website_sale.scss:77]: It allows the product image to fit its container, so the user can click everywhere on the frame to replace it. The spec has been added by sbu and the fix has been found by sad,
I will add that to the commit message

File: addons/website/static/src/js/tours/tour_utils.js
Comment (younn-o) [reply addons/website/static/src/js/tours/tour_utils.js:232]: I will drop that commit, I think that part has been taken care of in another PR

File: addons/website/views/snippets/s_carousel.xml
Comment (younn-o) [reply addons/website/views/snippets/s_carousel.xml:59]: keeping the </a> tag is possible but the `o_not_editable` class is also needed on the <span> children
I am changing to that :)

File: addons/website_sale/static/src/scss/website_sale.scss
Comment (qsm-odoo) [reply addons/website_sale/static/src/scss/website_sale.scss:80]: Ok, the `` tag is expanded but the size of the seen image is the same right?

Comment (younn-o) [reply addons/website_sale/static/src/scss/website_sale.scss:80]: exactly !

File: addons/website/views/snippets/s_carousel.xml
Comment (qsm-odoo) [reply addons/website/views/snippets/s_carousel.xml]: Yes, in stable, always better to minimize the XML changes to avoid breaking xpath or other custo

Comment (younn-o) [reply addons/website/views/snippets/s_carousel.xml]: Ok, thanks. In the end it was the contenteditable attribute that was mandatory on the `<a>` tag, nothing with the children

File: addons/website/views/snippets/s_carousel.xml
Comment (qsm-odoo) [inline addons/website/views/snippets/s_carousel.xml:59]: @younn-o You changed leaving the href... but you left the div... I guess it's a mistake? I'll try and fix on my own.

File: addons/web_editor/static/src/js/wysiwyg/widgets/media.js
Comment (qsm-odoo) [inline addons/web_editor/static/src/js/wysiwyg/widgets/media.js:954]: @younn-o I am moving this commit for 14.3 instead of 14.0 (not really stable, require rd-security approval, conflicts, etc)

File: addons/website_blog/views/snippets/snippets.xml
Comment (qsm-odoo) [inline addons/website_blog/views/snippets/snippets.xml:34]: @younn-o I am changing this to
```xml
<attribute name="data-exclude" add="#o_wblog_post_content *" separator=","/>
```
Otherwise you break potential custo by fully overriding the attribute

**Comment by qsm-odoo** [comment]
@robodoo rebase-ff r+

**Pull request closed without merging**
